### ISSUE TITLE
abort in fleece::Scope, on applicationWillTerminate with document lookup

### ISSUE THREAD

**Issue by iwasrobbed-ks** [couchbase/couchbase-lite-core#972]
Our largest crash seems to be one that only happens when an app is being backgrounded and the system calls `applicationWillTerminate` due to low device memory

We don't do anything in that delegate related to Couchbase.

Any ideas for what's happening?

#### Version

[CouchbaseLiteSwift v2.7.0](https://github.com/couchbase/couchbase-lite-ios/releases/tag/2.7.0) built as a framework

#### Crash:

```
std::__1::system_error: mutex lock failed: Invalid argument

0 unknown file unknown method
```

#### Stack trace:

I just uploaded the Couchbase dSYM to our crash reporter so we'll hopefully have better symbol names soon:

```
Thread 6
0 libsystem_kernel.dylib  ___psynch_cvwait
1 libsystem_pthread.dylib __pthread_cond_wait$VARIANT$armv81
2 libc++.1.dylib          std::__1::condition_variable::__do_timed_wait(std::__1::unique_lock<std::__1::mutex>&, std::__1::chrono::time_point<std::__1::chrono::system_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > >)
3 CouchbaseLiteSwift      litecore::actor::Timer::Manager::run()
4 CouchbaseLiteSwift      _hidden#7929_
5 CouchbaseLiteSwift      _hidden#7928_
6 CouchbaseLiteSwift      _hidden#7925_
7 libsystem_pthread.dylib __pthread_start
```

#### Code
Timer: https://github.com/couchbase/couchbase-lite-core/blob/master/LiteCore/Support/Timer.cc#L39

**Comment by borrrden** [comment]
@snej I am unable to make sense of why it would crash here....

**Comment by rob-keepsafe** [comment]
Might be similar to this: https://github.com/couchbase/couchbase-lite-core/issues/652#issuecomment-451322131

**Comment by borrrden** [comment]
This message is one of the most common errors that can happen with bad memory access.  It's basically no different than saying "EXC_BAD_ACCESS" or "segmentation fault".  I don't think it's similar to other ones that we've faced because of the stack trace being different.  It's very unusual to have it happen from the `run` method and I wonder if the trace is correct given the hidden frames.

**Comment by rob-keepsafe** [comment]
We'll try to get updated stack traces with our next release (in a week or two); we also have another larger crash that we need better stack traces on before I'll file an issue

**Comment by snej** [comment]
That backtrace of thread 6 above doesn't look as though it crashed ... it looks like it's still waiting on a condition variable. Are you sure it wasn't another thread that crashed? Could you upload the crash log?

**Comment by rob-keepsafe** [comment]
Going to close this as it's not reproing in our latest release and possibly was misattributed to Couchbase. I'll reopen if it resurfaces

**Comment by rob-keepsafe** [comment]
I finally had this happen to me in debug mode. Looks like the crash reporter misattributed it and it was happening on a different thread than the timer.

Effectively what's happening on our end is:
- app is backgrounded and the OS decides to terminate it
- we're listening for the replicator to stop so we know sync has stopped (see #917 for why)
- we try to use `database.document(withID: someId)`
- Couchbase hits this SIGABRT in a race condition w/ the app being terminated

>INFO [PVCouchbaseSync:305] | Replicator status: offline | queue: com.keepsafe.couchbaseSync.replicator
>
>libc++abi.dylib: terminating with uncaught exception of type std::__1::system_error: mutex lock failed: Invalid argument

<details>
<summary>Click to see threads</summary>
<pre>
(lldb) bt all
CouchbaseLiteSwift was compiled with optimization - stepping may behave oddly; variables may not be available.
/Users/rob/Library/Developer/Xcode/DerivedData/KeepSafe-brdvaarnweazhiezlzmopyareklh/Build/Products/Debug-iphonesimulator/KeepSafe.app/Frameworks/CouchbaseLiteSwift.framework/CouchbaseLiteSwift: Cannot load Swift type information; failed to initialize ClangImporter: error: virtual filesystem overlay file '/Users/couchbase/Library/Developer/Xcode/DerivedData/CouchbaseLite-fgsjkpbiwewkxtequbmsbskyiimb/Build/Intermediates.noindex/CouchbaseLite.build/Release-EE-iphonesimulator/CBL-EE Swift.build/all-product-headers.yaml' not found

  thread #1, queue = 'com.apple.main-thread'
    frame #0: 0x00007fff51b5d756 libsystem_kernel.dylib`__semwait_signal + 10
    frame #1: 0x00007fff51aed616 libsystem_c.dylib`nanosleep + 196
    frame #2: 0x00007fff51aed510 libsystem_c.dylib`usleep + 53
    frame #3: 0x00000001148f5b85 libclang_rt.tsan_iossim_dynamic.dylib`__tsan::Finalize(__tsan::ThreadState*) + 85
    frame #4: 0x00000001148e1b53 libclang_rt.tsan_iossim_dynamic.dylib`__tsan::finalize(void*) + 19
    frame #5: 0x00007fff51acfd51 libsystem_c.dylib`__cxa_finalize_ranges + 319
    frame #6: 0x00007fff51ad0076 libsystem_c.dylib`exit + 48
    frame #7: 0x00007fff48c90ce9 UIKitCore`-[UIApplication _terminateWithStatus:] + 276
    frame #8: 0x00007fff483928d9 UIKitCore`-[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:] + 111
    frame #9: 0x00007fff48392587 UIKitCore`-[_UISceneLifecycleMultiplexer forceExitWithTransitionContext:scene:] + 225
    frame #10: 0x00007fff48c8783d UIKitCore`-[UIApplication workspaceShouldExit:withTransitionContext:] + 166
    frame #11: 0x00007fff36ca6049 FrontBoardServices`-[FBSUIApplicationWorkspaceShim workspaceShouldExit:withTransitionContext:] + 79
    frame #12: 0x00007fff36cd2584 FrontBoardServices`__83-[FBSWorkspaceScenesClient willTerminateWithTransitionContext:withAcknowledgement:]_block_invoke_2 + 76
    frame #13: 0x00007fff36cb7757 FrontBoardServices`-[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] + 220
    frame #14: 0x00007fff36cd2514 FrontBoardServices`__83-[FBSWorkspaceScenesClient willTerminateWithTransitionContext:withAcknowledgement:]_block_invoke + 128
    frame #15: 0x0000000117d35e8e libdispatch.dylib`_dispatch_client_callout + 8
    frame #16: 0x0000000117d38da2 libdispatch.dylib`_dispatch_block_invoke_direct + 300
    frame #17: 0x00007fff36cf86e9 FrontBoardServices`__FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 30
    frame #18: 0x00007fff36cf83d7 FrontBoardServices`-[FBSSerialQueue _queue_performNextIfPossible] + 441
    frame #19: 0x00007fff36cf88e6 FrontBoardServices`-[FBSSerialQueue _performNextFromRunLoopSource] + 22
    frame #20: 0x00007fff23da0d31 CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
    frame #21: 0x00007fff23da0c5c CoreFoundation`__CFRunLoopDoSource0 + 76
    frame #22: 0x00007fff23da0434 CoreFoundation`__CFRunLoopDoSources0 + 180
    frame #23: 0x00007fff23d9b02e CoreFoundation`__CFRunLoopRun + 974
    frame #24: 0x00007fff23d9a944 CoreFoundation`CFRunLoopRunSpecific + 404
    frame #25: 0x00007fff38ba6c1a GraphicsServices`GSEventRunModal + 139
    frame #26: 0x00007fff48c8b9ec UIKitCore`UIApplicationMain + 1605
    frame #27: 0x000000010d541f4c KeepSafe`main at main.swift:33:1
    frame #28: 0x00007fff51a231fd libdyld.dylib`start + 1
    frame #29: 0x00007fff51a231fd libdyld.dylib`start + 1
  thread #6
    frame #0: 0x00007fff51b5d756 libsystem_kernel.dylib`__semwait_signal + 10
    frame #1: 0x00007fff51aed616 libsystem_c.dylib`nanosleep + 196
    frame #2: 0x00007fff51aed510 libsystem_c.dylib`usleep + 53
    frame #3: 0x0000000114900d98 libclang_rt.tsan_iossim_dynamic.dylib`__tsan::BackgroundThread(void*) + 424
    frame #4: 0x00007fff51c0c109 libsystem_pthread.dylib`_pthread_start + 148
    frame #5: 0x00007fff51c07b8b libsystem_pthread.dylib`thread_start + 15
  thread #7, name = 'com.apple.uikit.eventfetch-thread'
    frame #0: 0x00007fff51b5adfa libsystem_kernel.dylib`mach_msg_trap + 10
    frame #1: 0x00007fff51b5b170 libsystem_kernel.dylib`mach_msg + 60
    frame #2: 0x00007fff23da0595 CoreFoundation`__CFRunLoopServiceMachPort + 165
    frame #3: 0x00007fff23d9b1c7 CoreFoundation`__CFRunLoopRun + 1383
    frame #4: 0x00007fff23d9a944 CoreFoundation`CFRunLoopRunSpecific + 404
    frame #5: 0x00007fff25939c71 Foundation`-[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 211
    frame #6: 0x00007fff25939ee0 Foundation`-[NSRunLoop(NSRunLoop) runUntilDate:] + 72
    frame #7: 0x00007fff48d39bfb UIKitCore`-[UIEventFetcher threadMain] + 138
    frame #8: 0x00007fff2594f9eb Foundation`__NSThread__start__ + 1047
    frame #9: 0x00000001148c92cc libclang_rt.tsan_iossim_dynamic.dylib`__tsan_thread_start_func + 140
    frame #10: 0x00007fff51c0c109 libsystem_pthread.dylib`_pthread_start + 148
    frame #11: 0x00007fff51c07b8b libsystem_pthread.dylib`thread_start + 15
  thread #10, queue = 'SwiftWebSocket'
    frame #0: 0x00007fff51b5d882 libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007fff51c0c425 libsystem_pthread.dylib`_pthread_cond_wait + 698
    frame #2: 0x000000011490ea12 libclang_rt.tsan_iossim_dynamic.dylib`__tsan::call_pthread_cancel_with_cleanup(int (*)(void*, void*, void*), void*, void*, void*, void (*)(void*), void*) + 82
    frame #3: 0x00000001148c9c13 libclang_rt.tsan_iossim_dynamic.dylib`cond_wait(__tsan::ThreadState*, unsigned long, __tsan::ScopedInterceptor*, int (*)(void*, void*, void*), void*, void*, void*) + 243
    frame #4: 0x00000001148c9dc0 libclang_rt.tsan_iossim_dynamic.dylib`wrap_pthread_cond_timedwait + 336
    frame #5: 0x000000010e8e634b KeepSafe`Manager.wait(timeInMs=250, self=0x00007b1c00028630) at WebSocket.swift:1626:16
    frame #6: 0x000000010e8e4d32 KeepSafe`closure #1 in Manager.init(self=0x00007b1c00028630) at WebSocket.swift:1598:30
    frame #7: 0x000000010e8e909e KeepSafe`partial apply for closure #1 in Manager.init() at <compiler-generated>:0
    frame #8: 0x000000010cc9b0f4 KeepSafe`thunk for @escaping @callee_guaranteed () -> () at <compiler-generated>:0
    frame #9: 0x000000011490f53c libclang_rt.tsan_iossim_dynamic.dylib`__tsan::invoke_and_release_block(void*) + 12
    frame #10: 0x000000011490f292 libclang_rt.tsan_iossim_dynamic.dylib`__tsan::dispatch_callback_wrap(void*) + 306
    frame #11: 0x0000000117d35e8e libdispatch.dylib`_dispatch_client_callout + 8
    frame #12: 0x0000000117d3c6fd libdispatch.dylib`_dispatch_lane_serial_drain + 788
    frame #13: 0x0000000117d3d28f libdispatch.dylib`_dispatch_lane_invoke + 422
    frame #14: 0x0000000117d48b65 libdispatch.dylib`_dispatch_workloop_worker_thread + 719
    frame #15: 0x00007fff51c08a3d libsystem_pthread.dylib`_pthread_wqthread + 290
    frame #16: 0x00007fff51c07b77 libsystem_pthread.dylib`start_wqthread + 15
  thread #12, name = 'com.apple.NSURLConnectionLoader'
    frame #0: 0x00007fff51b5adfa libsystem_kernel.dylib`mach_msg_trap + 10
    frame #1: 0x00007fff51b5b170 libsystem_kernel.dylib`mach_msg + 60
    frame #2: 0x00007fff23da0595 CoreFoundation`__CFRunLoopServiceMachPort + 165
    frame #3: 0x00007fff23d9b1c7 CoreFoundation`__CFRunLoopRun + 1383
    frame #4: 0x00007fff23d9a944 CoreFoundation`CFRunLoopRunSpecific + 404
    frame #5: 0x00007fff23159df3 CFNetwork`___lldb_unnamed_symbol11344$$CFNetwork + 182
    frame #6: 0x00007fff2594f9eb Foundation`__NSThread__start__ + 1047
    frame #7: 0x00000001148c92cc libclang_rt.tsan_iossim_dynamic.dylib`__tsan_thread_start_func + 140
    frame #8: 0x00007fff51c0c109 libsystem_pthread.dylib`_pthread_start + 148
    frame #9: 0x00007fff51c07b8b libsystem_pthread.dylib`thread_start + 15
  thread #22, name = 'Timer (Couchbase Lite Core)'
    frame #0: 0x00007fff51b5d882 libsystem_kernel.dylib`__psynch_cvwait + 10
    frame #1: 0x00007fff51c0c425 libsystem_pthread.dylib`_pthread_cond_wait + 698
    frame #2: 0x000000011490ea12 libclang_rt.tsan_iossim_dynamic.dylib`__tsan::call_pthread_cancel_with_cleanup(int (*)(void*, void*, void*), void*, void*, void*, void (*)(void*), void*) + 82
    frame #3: 0x00000001148c9c13 libclang_rt.tsan_iossim_dynamic.dylib`cond_wait(__tsan::ThreadState*, unsigned long, __tsan::ScopedInterceptor*, int (*)(void*, void*, void*), void*, void*, void*) + 243
    frame #4: 0x00000001148c9dc0 libclang_rt.tsan_iossim_dynamic.dylib`wrap_pthread_cond_timedwait + 336
    frame #5: 0x00007fff4f99c743 libc++.1.dylib`std::__1::condition_variable::__do_timed_wait(std::__1::unique_lock<std::__1::mutex>&, std::__1::chrono::time_point<std::__1::chrono::system_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > >) + 93
    frame #6: 0x0000000115ac7633 CouchbaseLiteSwift`litecore::actor::Timer::Manager::run() [inlined] std::__1::cv_status std::__1::condition_variable::wait_for<long long, std::__1::ratio<1l, 1000000000l> >(this=<unavailable>, __lk=0x00007e8000abced0, __d=<unavailable>) at Timer.cc:0:22 [opt]
    frame #7: 0x0000000115ac762b CouchbaseLiteSwift`litecore::actor::Timer::Manager::run() [inlined] std::__1::cv_status std::__1::condition_variable::wait_until<std::__1::chrono::steady_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > >(this=<unavailable>, __lk=0x00007e8000abced0) at __mutex_base:383 [opt]
    frame #8: 0x0000000115ac762b CouchbaseLiteSwift`litecore::actor::Timer::Manager::run(this=0x00007b2400035f70) at Timer.cc:68 [opt]
    frame #9: 0x0000000115ac7513 CouchbaseLiteSwift`litecore::actor::Timer::Manager::Manager(this=<unavailable>)::$_0::operator()() const at Timer.cc:34:24 [opt]
    frame #10: 0x0000000115ac750a CouchbaseLiteSwift`decltype(__f=<unavailable>)::$_0>(fp)()) std::__1::__invoke<litecore::actor::Timer::Manager::Manager()::$_0>(litecore::actor::Timer::Manager::Manager()::$_0&&) at type_traits:4425:1 [opt]
    frame #11: 0x0000000115ac7447 CouchbaseLiteSwift`void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, litecore::actor::Timer::Manager::Manager()::$_0> >(void*) [inlined] void std::__1::__thread_execute<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, litecore::actor::Timer::Manager::Manager()::$_0>(__t=size=2)::$_0>&, std::__1::__tuple_indices<>) at thread:341:5 [opt]
    frame #12: 0x0000000115ac743e CouchbaseLiteSwift`void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, litecore::actor::Timer::Manager::Manager()::$_0> >(__vp=0x00007b0400014a60) at thread:351 [opt]
    frame #13: 0x00000001148c92cc libclang_rt.tsan_iossim_dynamic.dylib`__tsan_thread_start_func + 140
    frame #14: 0x00007fff51c0c109 libsystem_pthread.dylib`_pthread_start + 148
    frame #15: 0x00007fff51c07b8b libsystem_pthread.dylib`thread_start + 15
  thread #23, name = 'com.apple.CFStream.LegacyThread'
    frame #0: 0x00007fff51b5adfa libsystem_kernel.dylib`mach_msg_trap + 10
    frame #1: 0x00007fff51b5b170 libsystem_kernel.dylib`mach_msg + 60
    frame #2: 0x00007fff23da0595 CoreFoundation`__CFRunLoopServiceMachPort + 165
    frame #3: 0x00007fff23d9b1c7 CoreFoundation`__CFRunLoopRun + 1383
    frame #4: 0x00007fff23d9a944 CoreFoundation`CFRunLoopRunSpecific + 404
    frame #5: 0x00007fff23db8935 CoreFoundation`_legacyStreamRunLoop_workThread + 261
    frame #6: 0x00000001148c92cc libclang_rt.tsan_iossim_dynamic.dylib`__tsan_thread_start_func + 140
    frame #7: 0x00007fff51c0c109 libsystem_pthread.dylib`_pthread_start + 148
    frame #8: 0x00007fff51c07b8b libsystem_pthread.dylib`thread_start + 15
  thread #24, name = 'com.apple.CFSocket.private'
    frame #0: 0x00007fff51b630fe libsystem_kernel.dylib`__select + 10
    frame #1: 0x00007fff23da9cf4 CoreFoundation`__CFSocketManager + 660
    frame #2: 0x00000001148c92cc libclang_rt.tsan_iossim_dynamic.dylib`__tsan_thread_start_func + 140
    frame #3: 0x00007fff51c0c109 libsystem_pthread.dylib`_pthread_start + 148
    frame #4: 0x00007fff51c07b8b libsystem_pthread.dylib`thread_start + 15
  thread #25, name = 'AVAudioSession Notify Thread'
    frame #0: 0x00007fff51b5adfa libsystem_kernel.dylib`mach_msg_trap + 10
    frame #1: 0x00007fff51b5b170 libsystem_kernel.dylib`mach_msg + 60
    frame #2: 0x00007fff23da0595 CoreFoundation`__CFRunLoopServiceMachPort + 165
    frame #3: 0x00007fff23d9b1c7 CoreFoundation`__CFRunLoopRun + 1383
    frame #4: 0x00007fff23d9a944 CoreFoundation`CFRunLoopRunSpecific + 404
    frame #5: 0x00007fff20b109d2 AVFAudio`GenericRunLoopThread::Entry(void*) + 166
    frame #6: 0x00007fff20b4c06f AVFAudio`CAPThread::Entry(CAPThread*) + 191
    frame #7: 0x00000001148c92cc libclang_rt.tsan_iossim_dynamic.dylib`__tsan_thread_start_func + 140
    frame #8: 0x00007fff51c0c109 libsystem_pthread.dylib`_pthread_start + 148
    frame #9: 0x00007fff51c07b8b libsystem_pthread.dylib`thread_start + 15
  thread #26, name = 'AMCP Logging Spool'
    frame #0: 0x00007fff51b5ae36 libsystem_kernel.dylib`semaphore_wait_trap + 10
    frame #1: 0x00007fff4d9b4d48 caulk`caulk::mach::semaphore::wait() + 16
    frame #2: 0x00007fff4d9b02f2 caulk`caulk::semaphore::timed_wait(double) + 106
    frame #3: 0x00007fff4d9b39d4 caulk`caulk::concurrent::details::worker_thread::run() + 30
    frame #4: 0x00007fff4d9b3b24 caulk`void* caulk::thread_proxy<std::__1::tuple<caulk::thread::attributes, void (caulk::concurrent::details::worker_thread::*)(), std::__1::tuple<caulk::concurrent::details::worker_thread*> > >(void*) + 45
    frame #5: 0x00000001148c92cc libclang_rt.tsan_iossim_dynamic.dylib`__tsan_thread_start_func + 140
    frame #6: 0x00007fff51c0c109 libsystem_pthread.dylib`_pthread_start + 148
    frame #7: 0x00007fff51c07b8b libsystem_pthread.dylib`thread_start + 15
  thread #27
    frame #0: 0x00007fff51b5ae36 libsystem_kernel.dylib`semaphore_wait_trap + 10
    frame #1: 0x00007fff4d9b4d48 caulk`caulk::mach::semaphore::wait() + 16
    frame #2: 0x00007fff4d9b02f2 caulk`caulk::semaphore::timed_wait(double) + 106
    frame #3: 0x00007fff4d9b39d4 caulk`caulk::concurrent::details::worker_thread::run() + 30
    frame #4: 0x00007fff4d9b3b24 caulk`void* caulk::thread_proxy<std::__1::tuple<caulk::thread::attributes, void (caulk::concurrent::details::worker_thread::*)(), std::__1::tuple<caulk::concurrent::details::worker_thread*> > >(void*) + 45
    frame #5: 0x00000001148c92cc libclang_rt.tsan_iossim_dynamic.dylib`__tsan_thread_start_func + 140
    frame #6: 0x00007fff51c0c109 libsystem_pthread.dylib`_pthread_start + 148
    frame #7: 0x00007fff51c07b8b libsystem_pthread.dylib`thread_start + 15
  thread #35
    frame #0: 0x00007fff51b5c4ce libsystem_kernel.dylib`__workq_kernreturn + 10
    frame #1: 0x00007fff51c08aa1 libsystem_pthread.dylib`_pthread_wqthread + 390
    frame #2: 0x00007fff51c07b77 libsystem_pthread.dylib`start_wqthread + 15
  thread #36
    frame #0: 0x00007fff51c07b68 libsystem_pthread.dylib`start_wqthread
* thread #43, queue = 'com.keepsafe.couchbaseSync.replicator', stop reason = signal SIGABRT
    frame #0: 0x00007fff51b6133a libsystem_kernel.dylib`__pthread_kill + 10
    frame #1: 0x00007fff51c0be60 libsystem_pthread.dylib`pthread_kill + 430
    frame #2: 0x00000001148ccbc5 libclang_rt.tsan_iossim_dynamic.dylib`wrap_pthread_kill + 325
    frame #3: 0x00007fff51af0b7c libsystem_c.dylib`abort + 120
    frame #4: 0x00000001148cbf4c libclang_rt.tsan_iossim_dynamic.dylib`wrap_abort + 108
    frame #5: 0x00007fff4f9f7858 libc++abi.dylib`abort_message + 231
    frame #6: 0x00007fff4f9e8ca7 libc++abi.dylib`demangling_terminate_handler() + 238
    frame #7: 0x00007fff50ba8c13 libobjc.A.dylib`_objc_terminate() + 104
    frame #8: 0x00007fff4f9f6c87 libc++abi.dylib`std::__terminate(void (*)()) + 8
    frame #9: 0x00007fff4f9f6c29 libc++abi.dylib`std::terminate() + 41
    frame #10: 0x000000011597811b CouchbaseLiteSwift`__clang_call_terminate + 11
    frame #11: 0x0000000115aefe9b CouchbaseLiteSwift`fleece::impl::Scope::registr(this=<unavailable>) at Doc.cc:134:5 [opt]
    frame #12: 0x00000001159eced9 CouchbaseLiteSwift`litecore::VersionedDocument::VersDocScope::VersDocScope(this=0x00007b800004b000, fleeceData=<unavailable>, sk=<unavailable>, document_=0x00007b50000506b8) at VersionedDocument.hh:92:14 [opt]
    frame #13: 0x00000001159ecce1 CouchbaseLiteSwift`void std::__1::deque<litecore::VersionedDocument::VersDocScope, std::__1::allocator<litecore::VersionedDocument::VersDocScope> >::emplace_back<fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(fleece::alloc_slice const&, fleece::impl::SharedKeys*&&, litecore::VersionedDocument*&&) [inlined] litecore::VersionedDocument::VersDocScope::VersDocScope(this=<unavailable>, fleeceData=0x00007b5000050788, sk=<unavailable>, document_=<unavailable>) at VersionedDocument.hh:94:13 [opt]
    frame #14: 0x00000001159eccd9 CouchbaseLiteSwift`void std::__1::deque<litecore::VersionedDocument::VersDocScope, std::__1::allocator<litecore::VersionedDocument::VersDocScope> >::emplace_back<fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(fleece::alloc_slice const&, fleece::impl::SharedKeys*&&, litecore::VersionedDocument*&&) [inlined] void std::__1::allocator<litecore::VersionedDocument::VersDocScope>::construct<litecore::VersionedDocument::VersDocScope, fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(this=<unavailable>, __p=<unavailable>, __args=0x00007b5000050788, __args=0x00007e80007a8f08, __args=0x00007e80007a8f10) at memory:1825 [opt]
    frame #15: 0x00000001159eccd2 CouchbaseLiteSwift`void std::__1::deque<litecore::VersionedDocument::VersDocScope, std::__1::allocator<litecore::VersionedDocument::VersDocScope> >::emplace_back<fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(fleece::alloc_slice const&, fleece::impl::SharedKeys*&&, litecore::VersionedDocument*&&) [inlined] void std::__1::allocator_traits<std::__1::allocator<litecore::VersionedDocument::VersDocScope> >::__construct<litecore::VersionedDocument::VersDocScope, fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(__a=<unavailable>, __p=<unavailable>, __args=0x00007b5000050788, __args=0x00007e80007a8f08, __args=0x00007e80007a8f10) at memory:1717 [opt]
    frame #16: 0x00000001159eccd2 CouchbaseLiteSwift`void std::__1::deque<litecore::VersionedDocument::VersDocScope, std::__1::allocator<litecore::VersionedDocument::VersDocScope> >::emplace_back<fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(fleece::alloc_slice const&, fleece::impl::SharedKeys*&&, litecore::VersionedDocument*&&) [inlined] void std::__1::allocator_traits<std::__1::allocator<litecore::VersionedDocument::VersDocScope> >::construct<litecore::VersionedDocument::VersDocScope, fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(__a=<unavailable>, __p=<unavailable>, __args=0x00007b5000050788, __args=0x00007e80007a8f08, __args=0x00007e80007a8f10) at memory:1560 [opt]
    frame #17: 0x00000001159eccd2 CouchbaseLiteSwift`void std::__1::deque<litecore::VersionedDocument::VersDocScope, std::__1::allocator<litecore::VersionedDocument::VersDocScope> >::emplace_back<fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(this=0x00007b50000507b8 size=0, __args=0x00007b5000050788, __args=0x00007e80007a8f08, __args=0x00007e80007a8f10) at deque:1872 [opt]
    frame #18: 0x00000001159ecc59 CouchbaseLiteSwift`litecore::VersionedDocument::addScope(this=<unavailable>, body=0x00007b5000050788) at VersionedDocument.cc:92:27 [opt]
    frame #19: 0x00000001159eda05 CouchbaseLiteSwift`litecore::VersionedDocument::updateScope(this=<unavailable>) at VersionedDocument.cc:85:9 [opt]
    frame #20: 0x00000001159ed96c CouchbaseLiteSwift`litecore::VersionedDocument::decode(this=0x00007b50000506b8) at VersionedDocument.cc:65:9 [opt]
    frame #21: 0x00000001159ed92c CouchbaseLiteSwift`litecore::VersionedDocument::read(this=<unavailable>) [opt] [artificial]
    frame #22: 0x00000001159ed8f8 CouchbaseLiteSwift`litecore::VersionedDocument::VersionedDocument(this=0x00007b50000506b8, store=<unavailable>, docID=<unavailable>) at VersionedDocument.cc:37:9 [opt]
    frame #23: 0x0000000115a302aa CouchbaseLiteSwift`c4Internal::TreeDocument::TreeDocument(c4Internal::Database*, FLSlice) [inlined] litecore::VersionedDocument::VersionedDocument(this=<unavailable>, store=<unavailable>, docID=<unavailable>) at VersionedDocument.cc:36:5 [opt]
    frame #24: 0x0000000115a30299 CouchbaseLiteSwift`c4Internal::TreeDocument::TreeDocument(this=0x00007b5000050600, database=0x00007b4400009600, docID=<unavailable>) at TreeDocument.cc:45 [opt]
    frame #25: 0x0000000115a2e7bf CouchbaseLiteSwift`c4Internal::TreeDocumentFactory::newDocumentInstance(FLSlice) [inlined] c4Internal::TreeDocument::TreeDocument(this=<unavailable>, database=<unavailable>, docID=<unavailable>) at TreeDocument.cc:47:9 [opt]
    frame #26: 0x0000000115a2e7b1 CouchbaseLiteSwift`c4Internal::TreeDocumentFactory::newDocumentInstance(this=0x00007b0400017c90, docID=<unavailable>) at TreeDocument.cc:514 [opt]
    frame #27: 0x0000000115a066a5 CouchbaseLiteSwift`fleece::Retained<c4Internal::Document> fleece::function_ref<fleece::Retained<c4Internal::Document> ()>::callback_fn<c4doc_get::$_0>(long) [inlined] c4doc_get::$_0::operator(this=<unavailable>)() const at c4Document.cc:69:44 [opt]
    frame #28: 0x0000000115a06689 CouchbaseLiteSwift`fleece::Retained<c4Internal::Document> fleece::function_ref<fleece::Retained<c4Internal::Document> (callable=<unavailable>)>::callback_fn<c4doc_get::$_0>(long) at function_ref.hh:59 [opt]
    frame #29: 0x0000000115a06716 CouchbaseLiteSwift`C4Document* fleece::function_ref<C4Document* ()>::callback_fn<newDoc(bool, C4Error*, fleece::function_ref<fleece::Retained<c4Internal::Document> ()>)::$_17>(long) [inlined] fleece::function_ref<fleece::Retained<c4Internal::Document> ()>::operator(this=<unavailable>)() const at function_ref.hh:72:12 [opt]
    frame #30: 0x0000000115a06709 CouchbaseLiteSwift`C4Document* fleece::function_ref<C4Document* ()>::callback_fn<newDoc(bool, C4Error*, fleece::function_ref<fleece::Retained<c4Internal::Document> ()>)::$_17>(long) [inlined] newDoc(this=0x00007e80007a90b8)>)::$_17::operator()() const at c4Document.cc:53 [opt]
    frame #31: 0x0000000115a06706 CouchbaseLiteSwift`C4Document* fleece::function_ref<C4Document* ()>::callback_fn<newDoc(callable=139088228946104)>)::$_17>(long) at function_ref.hh:59 [opt]
    frame #32: 0x0000000115a067a7 CouchbaseLiteSwift`C4Document* c4Internal::tryCatch<C4Document*>(C4Error*, fleece::function_ref<C4Document* ()>) [inlined] fleece::function_ref<C4Document* ()>::operator(this=<unavailable>)() const at function_ref.hh:72:12 [opt]
    frame #33: 0x0000000115a067a2 CouchbaseLiteSwift`C4Document* c4Internal::tryCatch<C4Document*>(outError=0x00007e80007a9100, fn=<unavailable>)>) at c4ExceptionUtils.hh:60 [opt]
    frame #34: 0x0000000115998361 CouchbaseLiteSwift`-[CBLDocument initWithDatabase:documentID:includeDeleted:error:] [inlined] newDoc(mustExist=true, outError=0x00007e80007a9100)>) at c4Document.cc:52:12 [opt]
    frame #35: 0x000000011599834a CouchbaseLiteSwift`-[CBLDocument initWithDatabase:documentID:includeDeleted:error:] [inlined] c4doc_get(database=<unavailable>, docID=<unavailable>, mustExist=true, outError=0x00007e80007a9100) at c4Document.cc:68 [opt]
    frame #36: 0x000000011599830c CouchbaseLiteSwift`-[CBLDocument initWithDatabase:documentID:includeDeleted:error:](self=0x00007b1400034620, _cmd=<unavailable>, database=0x00007b2000056f00, documentID=<unavailable>, includeDeleted=NO, outError=0x0000000000000000) at CBLDocument.mm:98 [opt]
    frame #37: 0x000000011599cc50 CouchbaseLiteSwift`-[CBLDatabase documentWithID:error:](self=<unavailable>, _cmd=<unavailable>, documentID=<unavailable>, outError=0x0000000000000000) at CBLDatabase.mm:700:16 [opt]
    frame #38: 0x000000011599ab80 CouchbaseLiteSwift`-[CBLDatabase documentWithID:](self=<unavailable>, _cmd=<unavailable>, documentID=<unavailable>) at CBLDatabase.mm:176:12 [opt]
    frame #39: 0x000000011596c2b5 CouchbaseLiteSwift`Database.document(id=<unavailable>, self=0x00007b1000095700) at Database.swift:63:32 [opt]
    frame #40: 0x000000010d0d490e KeepSafe`createMainAlbumIfNeeded #1 (isDecoy=false, self=KeepSafe.PVAlbum, userSession=0x00007b2000056b80, database=0x00007b1000095700) in static PVAlbum.createMainAlbumsIfNeeded(_:) at PVAlbum.swift:348:28
    frame #41: 0x000000010d0d4ea7 KeepSafe`closure #1 in static PVAlbum.createMainAlbumsIfNeeded(self=KeepSafe.PVAlbum, userSession=0x00007b2000056b80, database=0x00007b1000095700) at PVAlbum.swift:356:17
    frame #42: 0x000000010d0d67c0 KeepSafe`partial apply for closure #1 in static PVAlbum.createMainAlbumsIfNeeded(_:) at <compiler-generated>:0
    frame #43: 0x000000011596de2b CouchbaseLiteSwift`partial apply for closure #1 in Database.inBatch(using:) [inlined] closure #1 (caught=nil) -> () in CouchbaseLiteSwift.Database.inBatch(using: () throws -> ()) throws -> () at Database.swift:210:21 [opt]
    frame #44: 0x000000011596de25 CouchbaseLiteSwift`partial apply for closure #1 in Database.inBatch(using:) at <compiler-generated>:0 [opt]
    frame #45: 0x000000011596de81 CouchbaseLiteSwift`partial apply for thunk for @callee_guaranteed () -> () [inlined] reabstraction thunk helper from @callee_guaranteed () -> () to @escaping @callee_guaranteed () -> () at <compiler-generated>:0 [opt]
    frame #46: 0x000000011596de7e CouchbaseLiteSwift`partial apply for thunk for @callee_guaranteed () -> () at <compiler-generated>:0 [opt]
    frame #47: 0x000000011596cc1e CouchbaseLiteSwift`thunk for @escaping @callee_guaranteed () -> () at <compiler-generated>:0 [opt]
    frame #48: 0x000000011599b772 CouchbaseLiteSwift`-[CBLDatabase inBatch:usingBlock:](self=<unavailable>, _cmd=<unavailable>, outError=0x00007e80007a98e0, block=<unavailable>) at CBLDatabase.mm:335:9 [opt]
    frame #49: 0x000000011596cb4c CouchbaseLiteSwift`Database.inBatch(block=<unavailable>, self=<unavailable>) at Database.swift:208:19 [opt]
    frame #50: 0x000000010d0d460c KeepSafe`static PVAlbum.createMainAlbumsIfNeeded(userSession=0x00007b2000056b80, self=KeepSafe.PVAlbum) at PVAlbum.swift:355:22
    frame #51: 0x000000010db92da5 KeepSafe`PVUserSession.createMainAlbumsIfNeeded(self=0x00007b2000056b80) at PVUserSession.swift:162:29
    frame #52: 0x000000010db93607 KeepSafe`PVUserSession.syncReplicatorStateDidChange(status=CouchbaseLiteSwift.Replicator.Status @ 0x00007e80007a9f90, self=0x00007b2000056b80) at PVUserSession.swift:134:13
    frame #53: 0x000000010db939ab KeepSafe`protocol witness for PVCouchbaseSyncDelegate.syncReplicatorStateDidChange(_:) in conformance PVUserSession at <compiler-generated>:0
    frame #54: 0x000000010d42bf19 KeepSafe`PVCouchbaseSync.replicatorDidChange(change=CouchbaseLiteSwift.ReplicatorChange @ 0x00007e80007aa7f0, self=0x00007b2800004880) at PVCouchbaseSync.swift:307:19
    frame #55: 0x000000010d42d807 KeepSafe`closure #4 in PVCouchbaseSync.startReplicationSync(change=CouchbaseLiteSwift.ReplicatorChange @ 0x00007e80007aa7f0, self=0x00007b08000ecce1) at PVCouchbaseSync.swift:270:28
    frame #56: 0x000000010d42d879 KeepSafe`partial apply for closure #4 in PVCouchbaseSync.startReplicationSync() at <compiler-generated>:0
  * frame #57: 0x0000000115950321 CouchbaseLiteSwift`closure #1 in Replicator.addChangeListener(change=<unavailable>, listener=0x000000010d42d850 KeepSafe`partial apply forwarder for closure #4 (CouchbaseLiteSwift.ReplicatorChange) -> () in KeepSafe.PVCouchbaseSync.(startReplicationSync in _1074188F918A9585CAC37EA93D1BBEBD)() -> () at <compiler-generated>, self=0x00007b0c0011d540) at Replicator.swift:135:13 [opt]
    frame #58: 0x000000011595037c CouchbaseLiteSwift`thunk for @escaping @callee_guaranteed (@guaranteed CBLReplicatorChange) -> () at <compiler-generated>:0 [opt]
    frame #59: 0x000000011490f53c libclang_rt.tsan_iossim_dynamic.dylib`__tsan::invoke_and_release_block(void*) + 12
    frame #60: 0x000000011490f292 libclang_rt.tsan_iossim_dynamic.dylib`__tsan::dispatch_callback_wrap(void*) + 306
    frame #61: 0x0000000117d35e8e libdispatch.dylib`_dispatch_client_callout + 8
    frame #62: 0x0000000117d3c6fd libdispatch.dylib`_dispatch_lane_serial_drain + 788
    frame #63: 0x0000000117d3d28f libdispatch.dylib`_dispatch_lane_invoke + 422
    frame #64: 0x0000000117d48b65 libdispatch.dylib`_dispatch_workloop_worker_thread + 719
    frame #65: 0x00007fff51c08a3d libsystem_pthread.dylib`_pthread_wqthread + 290
    frame #66: 0x00007fff51c07b77 libsystem_pthread.dylib`start_wqthread + 15
  thread #44
    frame #0: 0x00007fff51c07b68 libsystem_pthread.dylib`start_wqthread
  thread #47
    frame #0: 0x00007fff51c07b68 libsystem_pthread.dylib`start_wqthread
  thread #48
    frame #0: 0x00007fff51c07b68 libsystem_pthread.dylib`start_wqthread
  thread #49
    frame #0: 0x00007fff51c07b68 libsystem_pthread.dylib`start_wqthread
</pre>
</details>

Screenshots of the threads in case it's easier to read:

<img width="717" alt="Screen Shot 2020-06-05 at 10 26 14 AM" src="https://user-images.githubusercontent.com/30269720/83901770-fcb35c00-a6f6-11ea-889b-d91a56b5cf6a.png">

<img width="1792" alt="Screen Shot 2020-06-05 at 10 23 28 AM" src="https://user-images.githubusercontent.com/30269720/83901753-f58c4e00-a6f6-11ea-8c2e-bc1a6bb7f137.png">

<img width="845" alt="Screen Shot 2020-06-05 at 10 34 17 AM" src="https://user-images.githubusercontent.com/30269720/83901801-0a68e180-a6f7-11ea-9355-3f1f2f8c77e1.png">

<img width="1566" alt="Screen Shot 2020-06-05 at 10 34 30 AM" src="https://user-images.githubusercontent.com/30269720/83901812-0e94ff00-a6f7-11ea-91c0-15dbb69a958a.png">

<img width="933" alt="Screen Shot 2020-06-05 at 10 34 51 AM" src="https://user-images.githubusercontent.com/30269720/83901835-16ed3a00-a6f7-11ea-8e71-a4e161a91c18.png">

**Comment by snej** [comment]
Thanks! So the actual crash is in VersionedDocument. Weirdly, in your textual thread dump every other line is showing up blank, but I copied the stack frames from the source:
```
* thread #43, queue = 'com.keepsafe.couchbaseSync.replicator', stop reason = signal SIGABRT
    ...
    frame #10: __clang_call_terminate + 11
    frame #11: fleece::impl::Scope::registr(this=<unavailable>) at Doc.cc:134:5 [opt]
    frame #12: litecore::VersionedDocument::VersDocScope::VersDocScope(this=0x00007b800004b000, fleeceData=<unavailable>, sk=<unavailable>, document_=0x00007b50000506b8) at VersionedDocument.hh:92:14 [opt]
    frame #13: void std::__1::deque<litecore::VersionedDocument::VersDocScope, std::__1::allocator<litecore::VersionedDocument::VersDocScope> >::emplace_back<fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(fleece::alloc_slice const&, fleece::impl::SharedKeys*&&, litecore::VersionedDocument*&&) [inlined] litecore::VersionedDocument::VersDocScope::VersDocScope(this=<unavailable>, fleeceData=0x00007b5000050788, sk=<unavailable>, document_=<unavailable>) at VersionedDocument.hh:94:13 [opt]
    frame #14: void std::__1::deque<litecore::VersionedDocument::VersDocScope, std::__1::allocator<litecore::VersionedDocument::VersDocScope> >::emplace_back<fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(fleece::alloc_slice const&, fleece::impl::SharedKeys*&&, litecore::VersionedDocument*&&) [inlined] void std::__1::allocator<litecore::VersionedDocument::VersDocScope>::construct<litecore::VersionedDocument::VersDocScope, fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(this=<unavailable>, __p=<unavailable>, __args=0x00007b5000050788, __args=0x00007e80007a8f08, __args=0x00007e80007a8f10) at memory:1825 [opt]
    frame #15: void std::__1::deque<litecore::VersionedDocument::VersDocScope, std::__1::allocator<litecore::VersionedDocument::VersDocScope> >::emplace_back<fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(fleece::alloc_slice const&, fleece::impl::SharedKeys*&&, litecore::VersionedDocument*&&) [inlined] void std::__1::allocator_traits<std::__1::allocator<litecore::VersionedDocument::VersDocScope> >::__construct<litecore::VersionedDocument::VersDocScope, fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(__a=<unavailable>, __p=<unavailable>, __args=0x00007b5000050788, __args=0x00007e80007a8f08, __args=0x00007e80007a8f10) at memory:1717 [opt]
    frame #16: void std::__1::deque<litecore::VersionedDocument::VersDocScope, std::__1::allocator<litecore::VersionedDocument::VersDocScope> >::emplace_back<fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(fleece::alloc_slice const&, fleece::impl::SharedKeys*&&, litecore::VersionedDocument*&&) [inlined] void std::__1::allocator_traits<std::__1::allocator<litecore::VersionedDocument::VersDocScope> >::construct<litecore::VersionedDocument::VersDocScope, fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(__a=<unavailable>, __p=<unavailable>, __args=0x00007b5000050788, __args=0x00007e80007a8f08, __args=0x00007e80007a8f10) at memory:1560 [opt]
    frame #17: void std::__1::deque<litecore::VersionedDocument::VersDocScope, std::__1::allocator<litecore::VersionedDocument::VersDocScope> >::emplace_back<fleece::alloc_slice const&, fleece::impl::SharedKeys*, litecore::VersionedDocument*>(this=0x00007b50000507b8 size=0, __args=0x00007b5000050788, __args=0x00007e80007a8f08, __args=0x00007e80007a8f10) at deque:1872 [opt]
    frame #18: litecore::VersionedDocument::addScope(this=<unavailable>, body=0x00007b5000050788) at VersionedDocument.cc:92:27 [opt]
    frame #19: litecore::VersionedDocument::updateScope(this=<unavailable>) at VersionedDocument.cc:85:9 [opt]
    frame #20: litecore::VersionedDocument::decode(this=0x00007b50000506b8) at VersionedDocument.cc:65:9 [opt]
    frame #21: litecore::VersionedDocument::read(this=<unavailable>) [opt] [artificial]
    frame #22: litecore::VersionedDocument::VersionedDocument(this=0x00007b50000506b8, store=<unavailable>, docID=<unavailable>) at VersionedDocument.cc:37:9 [opt]
    frame #23: c4Internal::TreeDocument::TreeDocument(c4Internal::Database*, FLSlice) [inlined] litecore::VersionedDocument::VersionedDocument(this=<unavailable>, store=<unavailable>, docID=<unavailable>) at VersionedDocument.cc:36:5 [opt]
    ...
```

**Comment by snej** [comment]
BTW, Rob, are you a Couchbase customer? If so, this issue needs to go through your support engineer, not Github...

**Comment by snej** [comment]
The `terminate` call in `Scope::registr` is almost certainly because that method throws an exception despite being marked as `noexcept`. I did this deliberately so the exception wouldn't propagate; it's a fatal error, effectively the same as an assertion failure.

There should be a message logged about the exception ... is there any way to see that? (It's logged by libc++, not Couchbase, so it wouldn't go into our log file.)

**Comment by rob-keepsafe** [comment]
Thanks @snej ; now that I've hit this, it's an easy workaround on our side so pretty low priority to fix. I'll let you know if I can get a better exception for you, but the only thing showing in the logs is `libc++abi.dylib: terminating with uncaught exception of type std::__1::system_error: mutex lock failed: Invalid argument`

I'll reach out via support next time (just used to GitHub for open source)

**Comment by snej** [comment]
What's the workaround? It might give me a clue why this fails. Something's definitely going wrong in LiteCore.

If the exception logged really is about a mutex, then that's different than what I thought. Seems like it would have to be from corruption of the static sMutex ... weird.

**Comment by borrrden** [comment]
I am not seeing anything particularly actionable here.  Is there any insight on how to repro this?

**Comment by iwasrobbed-ks** [comment]
The only thing I can tell you is that I haven't seen this happen in iOS 14 yet. I'll close it out for now and reopen if we see an increase once we open up to more users

**Comment by RockyCognizant** [comment]
Yes, still crashes on iOS 16 with CouchbaseSwiftLite 3.0.2!

``` log
Hardware Model:      iPad11,4
Code Type:           ARM-64 (Native)
Role:                Foreground
Parent Process:      launchd [1]

Date/Time:           2023-05-09 22:35:55.5336 -0700
Launch Time:         2023-05-09 18:12:02.7785 -0700
OS Version:          iPhone OS 16.4.1 (20E252)
Release Type:        User

Exception Type:  EXC_BREAKPOINT (SIGTRAP)
Exception Codes: 0x0000000000000001, 0x00000001dd62ec24
Termination Reason: SIGNAL 5 Trace/BPT trap: 5
Terminating Process: exc handler [10182]

Triggered by Thread:  21

Application Specific Information:
abort() called

Thread 21 Crashed:
0   libsystem_c.dylib             	       0x1dd62ec24 __abort + 160
1   libsystem_c.dylib             	       0x1dd5d6184 abort + 192
2   libc++abi.dylib               	       0x234ecbbf8 abort_message + 132
3   libc++abi.dylib               	       0x234ebb444 demangling_terminate_handler() + 348
4   libobjc.A.dylib               	       0x1cf1f5ea4 _objc_terminate() + 144
5   CouchbaseLiteSwift            	       0x10b0bd2b4 fleece::Backtrace::installTerminateHandler(std::__1::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)>)::$_1::operator()() const::'lambda'()::operator()+ 2446004 () const + 404
6   CouchbaseLiteSwift            	       0x10b0bd120 fleece::Backtrace::installTerminateHandler(std::__1::function<void (std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)>)::$_1::operator()() const::'lambda'()::operator()+ 2445600 () const + 0
7   libc++abi.dylib               	       0x234ecafbc std::__terminate(void (*)()) + 16

```

**Comment by Vyazovoy** [comment]
Hi, @snej! I'm experiencing the same failure on Xcode 15 and CouchbaseLiteSwift 3.1.1
BTW, looks like [here](https://stackoverflow.com/a/66773373) is a good explanation of why it might happen. So maybe similar approach would help here.

**Comment by pasin** [comment]
I'm creating CBL-4959 to track the issue.
